assemblyscript/assemblyscript.go

```diff
@@ -0,0 +1,182 @@
+package assemblyscript
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"time"
+	"unicode/utf16"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+)
+
+// Instantiate instantiates a module implementing abort, trace, and seed for use from AssemblyScript programs.
+// The instantiated module will output abort messages to os.Stderr, not output trace messages, and use
+// crypto/rand as the source for seed values. If the AssemblyScript program is configured to use WASI, by
+// calling "import wasi" in any file, these functions will not be used.
+func Instantiate(ctx context.Context, r wazero.Runtime) (api.Closer, error) {
+	return NewModuleBuilder().Instantiate(ctx, r)
+}
+
+// ModuleBuilder allows configuring the module that will export functions used automatically by AssemblyScript.
+type ModuleBuilder interface {
+	// WithAbortWriter sets the io.Writer that should be used to write abort messages to.
+	// Abort messages are written when an error is thrown within an AssemblyScript program.
+	//
+	// Defaults to os.Stderr.
```

Comment:
so I probably caused this confusion earlier. I meant to default to ModuleConfig stderr, not os.stderr, as the former centralizes the sandbox config. So for example, if we set this manually here, this is set different than the module importing it. I think really the default should be AbortWriter -> importing ModuleConfig.Stderr (which is internally SysConfig). The only other option should be to turn it off entirely (WithAbortWriterDisabled())